File: sources/platforms/common/core/enum/Enum.cpp
#include "Enum.h"
#include "delegates.h"
#include "core/namespace/Namespace.h"
#include "core/function/Function.h"

TypescriptEnum::TypescriptEnum(const std::string& name, Environment* environment) : Base(name, environment) { }

TypescriptEnum::TypescriptEnum(const std::string& name, Base* space) : Base(name, space) { }

TypescriptEnum::~TypescriptEnum() { }

void TypescriptEnum::SetValue(const char* name, const long int& value)
{
	mValues[name] = value;
}

File: sources/unity/Assets/Plugins/General.Typescript/core/enum/Enum.cs
using System;
using System.Collections.Generic;

namespace General.Typescript
{
    public class Enum<T> : TypeInstance
    {
        private Dictionary<string, long> mValues = new Dictionary<string, long>();

        internal Enum(IntPtr context, IntPtr handle, Base parent) : base(context, handle, typeof(T), parent) { }

        public void SetValue(string name, long value)
        {
            Entry.Enum.SetValue(mHandle, name, value);
        }

        protected override void bind(string parentType)
        {
            Entry.Enum.Register(this);
        }
    }
}
